### PR TITLE
Fix sorting of groups and group members

### PR DESCRIPTION
**Pull request by jotjern**
I've added alphabetic sorting of groups based on short_name and sorting of group members based on, in order: sum of punishment values, first_name, last_name

### CODE REVIEW

File: frontend/src/helpers/sorting.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    return groups.sort((a, b) => a.name_short.localeCompare(b.name_short));
+}
+
+export function sortGroupUsers(members: GroupUser[], punishmentTypes: PunishmentType[]) {
+    const punishmentTypeMap = punishmentTypes.reduce((acc, curr) => {
+        acc[curr.punishment_type_id] = curr;
+        return acc;
+    }, {} as { [id: number]: PunishmentType });
+    return members.sort((a, b) => {
+        const punishment_value = (user: GroupUser) => user.punishments.reduce(
```
Comment (Terbau) [inline frontend/src/helpers/sorting.ts:13]: Pretty sure we talked about sorting this list based on how many punishments you currently have, not by the total value.

File: frontend/src/views/groups/GroupsView.tsx [hunk 2]
```diff
@@ -2,11 +2,12 @@ import { useQuery } from "@tanstack/react-query";
 import { Table } from "../../components/table";
 import { ME_URL, getGroupLeaderboardUrl } from "../../helpers/api";
 import axios, { AxiosResponse } from "axios";
-import { Group, User } from "../../helpers/types";
+import {Group, GroupUser, User} from "../../helpers/types";
```
Comment (Terbau) [inline frontend/src/views/groups/GroupsView.tsx:5]: Use prettier, please for the love of god.

**Review by Terbau — CHANGES_REQUESTED** [review]

File: frontend/src/helpers/sorting.ts (hunk 1 above)
Comment (jotjern) [reply frontend/src/helpers/sorting.ts:13]: true

File: frontend/src/views/groups/GroupsView.tsx (hunk 2 above)
Comment (jotjern) [reply frontend/src/views/groups/GroupsView.tsx]: yep

File: frontend/src/views/groups/GroupsView.tsx [hunk 3]
```diff
@@ -39,15 +49,21 @@ export const GroupsView = () => {
     queryFn: () =>
       axios
         .get(getGroupLeaderboardUrl(selectedGroup!.group_id))
-        .then((res: AxiosResponse<Group>) => res.data),
+        .then((res: AxiosResponse<Group>) => {
+          const group = res.data;
+          group.members = sortGroupUsers(group.members);
```
Comment (Terbau) [inline frontend/src/views/groups/GroupsView.tsx:54]: Did you forget to pass `group.punishment_types` to the function?

**Review by Terbau — CHANGES_REQUESTED** [review]

**Review by Terbau — APPROVED** [review]
lgtm

**Pull request merged**